### PR TITLE
frontend: utils: Use parseFloat in compareUnits to support decimals and add test coverage

### PR DESCRIPTION
**Pull request by Suyog241005**
## Summary

This PR updates `compareUnits` in `util.ts` to use `parseFloat` instead of `parseInt`. This ensures that fractional/decimal values are compared accurately. In addition, it adds comprehensive unit test coverage.

## Changes

- Modified `compareUnits` in `frontend/src/lib/util.ts` to parse values using `Number.parseFloat` instead of `parseInt`, with added safety checks against `NaN`.
- Updated `util.test.ts` to verify the new behavior, including checks for matching and mismatching decimal values.

## Steps to Test

Run the frontend utility tests to verify they all pass:
1. `cd frontend`
2. `npm run test src/lib/util.test.ts`

## Screenshots (if applicable)

*N/A (Utility/logic unit tests)*

## Notes for the Reviewer

Previously, `compareUnits` used `parseInt` which caused fractional quantities to lose their decimal precision (e.g., comparing `1.5Gi` and `1.61 GB` incorrectly returned `true` because both parsed as `1`). In the ResourceQuota details view, this would incorrectly collapse/hide the original input value (rendering only `1.61 GB`). 
With `parseFloat`, these comparisons are accurate (they now evaluate to `false`), allowing the UI to correctly display both values, e.g., `1.5Gi (1.61 GB)`.

### CODE REVIEW

**Comment by Suyog241005** [comment]
Hi @yolossn & @vyncent-t  ,

I have added unit test coverage for the `compareUnits` helper function inside `util.ts` (`frontend/src/lib/util.test.ts`). The tests successfully run and verify correctness across edge cases (identical values, whitespace handling, case insensitivity, number-parsing matches, and mismatched inputs).

**Comment by Suyog241005** [comment]
Hi @illume , I have addressed the copilot suggestions.

**Review by illume — CHANGES_REQUESTED** [review]
Thanks for these changes.


Can you please address the open review comments? Once you've resolved each one, please mark it as resolved.

**Comment by Suyog241005** [comment]
Hi @illume,
I have addressed the copilot's comments.

**Comment by Suyog241005** [comment]
Hi @illume , a gentle follow up on this PR

**Comment by Suyog241005** [comment]
Hi @illume ,

This storybook snapshot test failure is unrelated to this PR, as this branch only contains changes to the utility files (`util.ts` and `util.test.ts`). 

```js
⎯⎯⎯⎯⎯⎯⎯ Failed Tests 3 ⎯⎯⎯⎯⎯⎯⎯

 FAIL  src/storybook.test.tsx > Storybook Tests > Job/JobDetailsView > Default
Error: Snapshot `Storybook Tests > Job/JobDetailsView > Default 1` mismatched

- Expected
+ Received

@@ -688,11 +688,23 @@
                    >
                      Events
                    </h2>
                    <div
                      class="MuiBox-root css-ldp2l3"
-                   />
+                   >
+                     <button
+                       aria-label="Events lifetime information"
+                       class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-ai5t7w-MuiButtonBase-root-MuiIconButton-root"
+                       data-mui-internal-clone-element="true"
+                       tabindex="0"
+                       type="button"
+                     >
+                       <span
+                         class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
+                       />
+                     </button>
+                   </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-1txv3mw"

 ❯ src/storybook.test.tsx:245:11
    243|           document.body.removeAttribute('style');
    244| 
    245|           await expect(document.body).toMatchFileSnapshot(snapshotPath…
       |           ^
    246|         });
    247|       });

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/3]⎯

 FAIL  src/storybook.test.tsx > Storybook Tests > Job/JobDetailsView > Running
Error: Snapshot `Storybook Tests > Job/JobDetailsView > Running 1` mismatched

- Expected
+ Received

@@ -593,11 +593,23 @@
                    >
                      Events
                    </h2>
                    <div
                      class="MuiBox-root css-ldp2l3"
-                   />
+                   >
+                     <button
+                       aria-label="Events lifetime information"
+                       class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-ai5t7w-MuiButtonBase-root-MuiIconButton-root"
+                       data-mui-internal-clone-element="true"
+                       tabindex="0"
+                       type="button"
+                     >
+                       <span
+                         class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
+                       />
+                     </button>
+                   </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-1txv3mw"

 ❯ src/storybook.test.tsx:245:11
    243|           document.body.removeAttribute('style');
    244| 
    245|           await expect(document.body).toMatchFileSnapshot(snapshotPath…
       |           ^
    246|         });
    247|       });

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[2/3]⎯

 FAIL  src/storybook.test.tsx > Storybook Tests > VolumeAttributesClass/DetailsView > Base
Error: Snapshot `Storybook Tests > VolumeAttributesClass/DetailsView > Base 1` mismatched

- Expected
+ Received

@@ -218,11 +218,23 @@
                    >
                      Events
                    </h2>
                    <div
                      class="MuiBox-root css-ldp2l3"
-                   />
+                   >
+                     <button
+                       aria-label="Events lifetime information"
+                       class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-ai5t7w-MuiButtonBase-root-MuiIconButton-root"
+                       data-mui-internal-clone-element="true"
+                       tabindex="0"
+                       type="button"
+                     >
+                       <span
+                         class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
+                       />
+                     </button>
+                   </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-1txv3mw"

 ❯ src/storybook.test.tsx:245:11
    243|           document.body.removeAttribute('style');
    244| 
    245|           await expect(document.body).toMatchFileSnapshot(snapshotPath…
       |           ^
    246|         });
    247|       });

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[3/3]⎯


  Snapshots  3 failed
 Test Files  1 failed | 122 passed (123)
      Tests  3 failed | 1775 passed (1778)
   Start at  14:45:00
   Duration  92.40s (transform 18.32s, setup 6.85s, collect 113.74s, tests 76.14s, environment 32.69s, prepare 7.69s)


Error: Error: Snapshot `Storybook Tests > Job/JobDetailsView > Default 1` mismatched

- Expected
+ Received

@@ -688,11 +688,23 @@
                    >
                      Events
                    </h2>
                    <div
                      class="MuiBox-root css-ldp2l3"
-                   />
+                   >
+                     <button
+                       aria-label="Events lifetime information"
+                       class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-ai5t7w-MuiButtonBase-root-MuiIconButton-root"
+                       data-mui-internal-clone-element="true"
+                       tabindex="0"
+                       type="button"
+                     >
+                       <span
+                         class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
+                       />
+                     </button>
+                   </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-1txv3mw"

 ❯ src/storybook.test.tsx:245:11



Error: Error: Snapshot `Storybook Tests > Job/JobDetailsView > Running 1` mismatched

- Expected
+ Received

@@ -593,11 +593,23 @@
                    >
                      Events
                    </h2>
                    <div
                      class="MuiBox-root css-ldp2l3"
-                   />
+                   >
+                     <button
+                       aria-label="Events lifetime information"
+                       class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-ai5t7w-MuiButtonBase-root-MuiIconButton-root"
+                       data-mui-internal-clone-element="true"
+                       tabindex="0"
+                       type="button"
+                     >
+                       <span
+                         class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
+                       />
+                     </button>
+                   </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-1txv3mw"

 ❯ src/storybook.test.tsx:245:11



Error: Error: Snapshot `Storybook Tests > VolumeAttributesClass/DetailsView > Base 1` mismatched

- Expected
+ Received

@@ -218,11 +218,23 @@
                    >
                      Events
                    </h2>
                    <div
                      class="MuiBox-root css-ldp2l3"
-                   />
+                   >
+                     <button
+                       aria-label="Events lifetime information"
+                       class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-ai5t7w-MuiButtonBase-root-MuiIconButton-root"
+                       data-mui-internal-clone-element="true"
+                       tabindex="0"
+                       type="button"
+                     >
+                       <span
+                         class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
+                       />
+                     </button>
+                   </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-1txv3mw"

 ❯ src/storybook.test.tsx:245:11


make: *** [Makefile:353: frontend-test] Error 1
```

**Comment by Suyog241005** [comment]
Hi @illume , I have addressed the copilot's comments.

**Comment by Suyog241005** [comment]
hi @illume ,I have rebased the branch

**Comment by Suyog241005** [comment]
Hi @illume , I have addressed the copilot's comments.

**Comment by Suyog241005** [comment]
Hi @illume ,  I think this PR is ready to be merged.

**Review by illume — APPROVED** [review]
🎉 thanks!
